File: 28.ImplementstrStr/strStr.cc
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    int strStr(string haystack, string needle) {
        if (haystack == "" && needle == "") {
            return 0;
        }

        for (int i = 0; i < int(haystack.size()) - int(needle.size()) + 1; i++) {
            cout << i << endl;
            if (haystack.substr(i, needle.size()) == needle) {
                return i;
            }
        }

        return -1;
    }
};

int main(int argc, char const *argv[]) {
    string haystack = "abb";
    string needle = "abaaa";
    Solution s = Solution();
    cout << s.strStr(haystack, needle) << endl;
    return 0;
}

File: 28.ImplementstrStr/strStr.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        for i in range(len(haystack) - len(needle) + 1):
            if haystack[i:i+len(needle)] == needle:
                return i

        return -1



if __name__ == '__main__':
    haystack = ""#'this is a book'
    needle = ""#'isa'
    print Solution().strStr(haystack, needle)
